components/ntp_background_images/browser/ntp_background_images_utils.cc

```diff
@@ -5,13 +5,44 @@
 
 #include "brave/components/ntp_background_images/browser/ntp_background_images_utils.h"
 
+#include <string>
+
+#include "base/files/file_path.h"
+#include "brave/components/ntp_background_images/browser/ntp_background_images_data.h"
 #include "brave/components/ntp_background_images/common/pref_names.h"
-#include "components/pref_registry/pref_registry_syncable.h"
+#include "components/prefs/pref_registry_simple.h"
 
 namespace ntp_background_images {
 
-void RegisterProfilePrefs(user_prefs::PrefRegistrySyncable* registry) {
-  registry->RegisterBooleanPref(prefs::kNewTabPageShowBackgroundImage, true);
+void RegisterLocalStatePrefs(PrefRegistrySimple* registry) {
+  registry->RegisterStringPref(prefs::kNewTabPageCachedReferralPromoCode,
+                               std::string());
+  registry->RegisterDictionaryPref(
+      prefs::kNewTabPageCachedSuperReferrerComponentInfo);
+}
+
+std::unique_ptr<NTPBackgroundImagesData> GetDemoWallpaper(bool super_referrer) {
```

Comment:
we need to rethink this demo wallpaper thing because it's not usable on android and having a different url path makes things unnecessarily complicated. I think the demo should just be another component id

Reply:
Good point with demo - maybe that can be the same as hardcoding `promoCode` file as `TECHNIK`

Reply:
Yes, using component for demo is good idea. and also using static string for super referrer demo is good idea.

Reply:
Demo data is also provided by real component instead of using static assets.